server/enterprise/message_export/actiance_export/actiance_export.go

```diff
@@ -224,6 +234,63 @@ func ActianceExport(rctx request.CTX, p Params) (shared.RunExportResults, error)
 	return results, err
 }
 
+func getPostExport(posts []*model.MessageExport, i int, results shared.RunExportResults) (PostExport, shared.RunExportResults) {
+	// We have three "kinds" of posts:
+	// (using "1" and "2" for simplicity)
+	// - created:                         Id = new,  CreateAt = 1,    UpdateAt = 1, DeleteAt = 0
+	// - deleted:                         Id = orig, CreateAt = orig, UpdateAt = 2, DeleteAt = 2, props: deleteBy
+	// - edited: old post gets "created": Id = new,  CreateAt = 1,    UpdateAt = 2, DeleteAt = 2, originalId: orig
+	//           existing post modified:  Id = orig, CreateAt = 1,    UpdateAt = 2, DeleteAt = 0
+	//
+	// We also have other ways for a post to be updated:
+	//  - a root post in a thread is replied to, when a reply is edited, or (as of 10.2) when a reply is deleted
+
+	post := posts[i]
+	if isEditedOriginalMsg(post) {
+		// Post has been edited. This is the original message.
+		results.EditedOrigMsgPosts++
+		return editedOriginalMsgToExportEntry(post), results
+	} else if isDeletedMsg(post) {
+		// Post is deleted
+		results.DeletedPosts++
+		return deletedPostToExportEntry(post, "delete "+*post.PostMessage), results
+	} else if *post.PostUpdateAt > *post.PostCreateAt {
+		// Post has been updated. But what kind?
+		if post.PostEditAt != nil && *post.PostEditAt > 0 {
+			// This is an edited post.
+			results.EditedNewMsgPosts++
+			return editedNewMsgToExportEntry(post), results
+		}
+		// This is just an updated post (e.g. reaction)
+		results.UpdatedPosts++
+		return updatedPostToExportEntry(post), results
+	}
+	// Post is newly created:
+	// *post.PostCreateAt == *post.PostUpdateAt && (post.PostDeleteAt == nil || *post.PostDeleteAt == 0)
+	// but also fallback to this in case there is missing data, which is better than not exporting anything.
+	results.CreatedPosts++
+	return createdPostToExportEntry(post), results
+}
+
+func isEditedOriginalMsg(post *model.MessageExport) bool {
+	return post.PostDeleteAt != nil && *post.PostDeleteAt > 0 && post.PostOriginalId != nil && *post.PostOriginalId != ""
+}
+
+func isDeletedMsg(post *model.MessageExport) bool {
+	if post.PostDeleteAt != nil && *post.PostDeleteAt > 0 && post.PostProps != nil {
+		props := map[string]any{}
+		err := json.Unmarshal([]byte(*post.PostProps), &props)
+		if err != nil {
+			return false
+		}
+
+		if _, ok := props[model.PostPropsDeleteBy]; ok {
```

Comment:
The `model.PostPropsDeleteBy` post prop? In `SqlPostStore.Delete`.

Reply:
Could this prop be tampered with by a malicious client?

Reply:
Yes, as discussed in our call. I'll make a post in the security channel with the findings from my investigation.

As for what would get exported, if there is no `deletedBy` prop, then this won't be considered a deleted post. If it was tampered with, then it will have an `updatedAt` > `createdAt` and so it will be considered an updated post (like a reaction). So we'll output it, but it won't be as a deleted post. 

I don't think we should make logic to deal with this because it's not something we should be expected to cover -- we would also have to start looking at all the combinations of post props and post fields and handle any combination that could be created through the api. And then maybe through store calls as well.